### PR TITLE
Add gradle dependency check task & upgrade dependencies

### PR DESCRIPTION
**Pull request by kingthorin**
- Similar to: https://github.com/zaproxy/zap-extensions/pull/2308 `./gradlew dependencyUpdates`, should appear under the "help" Gradle Tasks grouping in Eclipse.
- Update most dependencies, with minor code tweaks to address moved or deprecated classes/functionality.

Signed-off-by: kingthorin <kingthorin@users.noreply.github.com>

### CODE REVIEW

**Comment by kingthorin** [comment]
So things worked fine in Travis for Java 11, and locally (with 11). What do I do about:
```
> Task :zap:compileJava
/home/travis/build/zaproxy/zaproxy/zap/src/main/java/org/zaproxy/zap/control/AddOn.java:50: error: cannot access CycleDetector
import org.jgrapht.alg.cycle.CycleDetector;
                            ^
  bad class file: /home/travis/.gradle/caches/modules-2/files-2.1/org.jgrapht/jgrapht-core/1.5.0/822a36ce09237f067a0278563e642af631e502ee/jgrapht-core-1.5.0.jar(org/jgrapht/alg/cycle/CycleDetector.class)
    class file has wrong version 55.0, should be 52.0
    Please remove or make sure it appears in the correct subdirectory of the classpath.
/home/travis/build/zaproxy/zaproxy/zap/src/main/java/org/zaproxy/zap/control/AddOn.java:237: error: cannot find symbol
    private Version version;
            ^
```

**Comment by thc202** [comment]
That requires a newer Java version.

File: zap/zap.gradle.kts [hunk 1]
```diff
@@ -41,56 +41,56 @@ tasks.named<JacocoReport>("jacocoTestReport") {
 }
 
 dependencies {
-    api("com.fifesoft:rsyntaxtextarea:3.0.4")
+    api("com.fifesoft:rsyntaxtextarea:3.1.1")
     api("com.github.zafarkhaja:java-semver:0.9.0")
     api("commons-beanutils:commons-beanutils:1.9.4")
-    api("commons-codec:commons-codec:1.13")
-    api("commons-collections:commons-collections:3.2.2")
```
Comment (thc202) [inline zap/zap.gradle.kts:48]: This needs to be kept, unless we want to stop distribute it in core. We could start by changing it to `runtimeOnly` (or just remove it, but we should quick check that we are not breaking known add-ons).

File: zap/zap.gradle.kts [hunk 2]
```diff
@@ -41,56 +41,56 @@ tasks.named<JacocoReport>("jacocoTestReport") {
...
     api("net.htmlparser.jericho:jericho-html:3.4")
     api("net.sf.json-lib:json-lib:2.4:jdk15")
-    api("org.apache.commons:commons-csv:1.7")
-    api("org.bouncycastle:bcmail-jdk15on:1.64")
-    api("org.bouncycastle:bcprov-jdk15on:1.64")
-    api("org.bouncycastle:bcpkix-jdk15on:1.64")
-    api("org.hsqldb:hsqldb:2.5.0")
-    api("org.jfree:jfreechart:1.0.19")
-    api("org.jgrapht:jgrapht-core:0.9.0")
+    api("org.apache.commons:commons-collections4:4.4")
```
Comment (thc202) [inline zap/zap.gradle.kts:59]: We should not introduce it as `api`, it does not seem to be needed though.

File: zap/zap.gradle.kts (hunk 1 above)
Comment (kingthorin) [reply zap/zap.gradle.kts:48]: I guess we have to keep it for now, it seems used by ascanrules.

File: zap/zap.gradle.kts (hunk 2 above)
Comment (kingthorin) [reply zap/zap.gradle.kts:59]: Reverted.

File: zap/zap.gradle.kts (hunk 1 above)
Comment (thc202) [reply zap/zap.gradle.kts:48]: Sounds good.

**Comment by thc202** [comment]
`LEGALNOTICE` should be updated.

**Comment by kingthorin** [comment]
Done

**Comment by thc202** [comment]
Missing tweak in BouncyCastle libs.

**Comment by kingthorin** [comment]
I searched for "bouncy" :(

**Comment by kingthorin** [comment]
Tweaked

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thanks!

**Review by psiinon — APPROVED** [review]

**Pull request merged**